Added compression parameters selection to CLI

ni4 opened the pull request:
- Added --zlib, --zip and --bzip2, -z 0..9 parameters to the rnp commandline and implemented their handling. Since compression now is used only during encryption it works only in this case.
- Fixed hanging bug in bzip2 decompression
- Updated CLI tests with more compression/decompression cases
- (hopefully) Fixed possible random CLI tests crash in Travis
- fixes #359

ni4 commented:
Looks like I'm lucky enough to hit #400
And another check hanged during the detached signature validation

dewyatt commented on src/tests/cli_tests.py:
```diff
@@ -144,18 +152,44 @@ def raise_err(msg, log = None):
         print log
     raise NameError(msg)
```
Just a nitpick but `NameError` isn't really an appropriate error to raise here. Best practice would be to add a small class that derives from `Exception` and takes that optional log parameter (and override `__str__` to print that out if provided).

ni4 replied:
Thanks, @dewyatt . It was just a quick pick, will fix it to work in a better way at some point.

ni4 commented:
I fixed hang during the signature verification, however still have #400 in one of the checks. 
Also updated exception throwing according to @dewyatt 's suggestion.
It seems that it would be a good idea to show less output with --debug so more tests could be run.

dewyatt commented:
> It seems that it would be a good idea to show less output with --debug so more tests could be run.

I don't think we're anywhere near the travis limits on this, and the debug output really does help so I would prefer we keep it. You may hit the limit on the web interface, but you can always download the raw log (click on "Raw Log").

Some other minor suggestions for later:

* You may be interested in using the `--with-colons` option for `gpg --list-keys`, very easy to parse in a robust manner.
* It would be nice if we added a `prefix=rnp-cli` (or similar) to the `mkdtemp` call. Otherwise, when debugging, we end up with a bunch of directory names like `/tmp/tmpw0nT6G/` which are pretty generic sounding (it feels safer to `rm -rf /tmp/rnp-*`, rather than `rm -rf /tmp/tmp*`).

ronaldtse commented:
Agree with both suggestions from @dewyatt . The last one is super easy to change.

@ni4 look like we can we rebase this and merge? This will solve #359

ni4 commented:
@ronaldtse I fixed prefix for mkdtemp, and now we know why tests are failing (see PR #429 ). So now this PR can be safely rebased and merged. Or should I do this?

ronaldtse commented:
Thanks for finding that issue out with the RSA keys. @ni4 it would be great if you could rebase and merge it 👍

ni4 commented:
@ronaldtse Done, except I forgot to squash, sorry.

ronaldtse commented:
No problem at all @ni4, thanks!
